### DISCUSSION TITLE
Correlation antar data di dataframe

### DISCUSSION THREAD

**Discussion by adiperdana23** [teamalgoritma/community#52] [Q&A] [answered]
Halo, saya ingin menanyakan bgmn mencari correlation antar data di kolom  pada data data frame, contoh pada data retail 
[retail.csv](https://github.com/teamalgoritma/community/files/8504091/retail.csv)

**Comment by tomytjandra** [comment, marked as answer]
Halo Pak @adiperdana23 untuk menghitung nilai correlation antar dua kolom numerik kita dapat menggunakan fungsi `cor()` pada base R.

**Definisi:** Correlation menunjukkan seberapa kuat hubungan antara dua variabel numerik. Range nilai correlation adalah dari -1 sampai 1
- Positif (searah): Ketika satu variabel nilainya bertambah, variabel satunya lagi bertambah, begitu pula sebaliknya.
- Negatif (berbanding terbalik): Ketika satu variabel nilainya bertambah, variabel satunya lagi berkurang, begitu pula sebaliknya.
- Nol: tidak ada korelasi

**Contoh:** Ingin menghitung nilai correlation antara nilai `Sales` dan `Profit` pada data `retail`

```
retail <- read.csv("retail.csv")
cor(retail$Sales, retail$Profit)
```

**Interpretasi:** Ternyata hasil korelasinya adalah 0.48, maka hubungan antara `Sales` dan `Profit` positif (searah) dan kekuatannya biasa saja.

**Penekanan:** Saat kita menggunakan nilai korelasi, tidak boleh menggunakan kata-kata "mempengaruhi" atau "dependent". Beberapa contoh visualisasi yang mendukung pernyataan "korelasi tidak menunjukkan hubungan sebab-akibat": https://www.tylervigen.com/spurious-correlations

**Tambahan:** Ketika kita ingin menampilkan nilai korelasi lebih dari 2 kolom numerik, bisa menggunakan fungsi `ggcorr()` dari library `GGally`

```
library(GGally)
ggcorr(retail[, c("Sales", "Quantity", "Discount", "Profit")],
       label = TRUE)
```

![image](https://user-images.githubusercontent.com/48079239/163772019-07c64f15-e086-47e4-a707-46a8f451a848.png)
